src/core/scala/app/com/linkedin/playparseq/s/stores/ParSeqTaskStore.scala

```diff
@@ -62,41 +69,29 @@ class ParSeqTaskStoreImpl extends ParSeqTaskStore {
   /**
    * The field ArgumentsKey is the default key of ParSeq Tasks.
    */
-  val ArgumentsKey = "ParSeqTasks"
+  val ArgumentsKey: TypedKey[MutableSet[Task[_]]] = TypedKey("ParSeqTasks")
 
   /**
    * @inheritdoc
    */
-  override def put(task: Task[_])(implicit requestHeader: RequestHeader): Unit = {
-    requestHeader match {
-      case _: ContextRequest[_] => get.add(task)
-      case _ =>
-    }
-  }
+  override def put(task: Task[_])(implicit requestHeader: RequestHeader): Unit = getOption.map(_.add(task))
 
   /**
    * @inheritdoc
    */
-  override def get(implicit requestHeader: RequestHeader): mutable.Set[Task[_]] = {
-    requestHeader match {
-      case ctx: ContextRequest[_] => ctx.args.get(ArgumentsKey).map(_.asInstanceOf[mutable.Set[Task[_]]])
-        .getOrElse({
-        val tasks: mutable.Set[Task[_]] = Collections.newSetFromMap[Task[_]](new ConcurrentHashMap).asScala
-        ctx.args.putIfAbsent(ArgumentsKey, tasks).getOrElse(tasks).asInstanceOf[mutable.Set[Task[_]]]
-      })
-      case _ => mutable.Set.empty
-    }
-  }
+  override def get(implicit requestHeader: RequestHeader): Set[Task[_]] = getOption.map(_.toSet).getOrElse(Set.empty)
 
-}
+  /**
+   * @inheritdoc
+   */
+  override def initialize[T](request: Request[T]): Request[T] = request.addAttr(ArgumentsKey, MutableSet.empty[Task[_]])
```

Comment:
The Set here needs to be threadsafe right?

Reply:
Good catch! Fixed!